Parser/string_parser.c

```diff
@@ -41,8 +41,40 @@ warn_invalid_escape_sequence(Parser *p, const char *first_invalid_escape, Token
     else {
         category = PyExc_DeprecationWarning;
     }
+
+    // Calculate the lineno and the col_offset of the invalid escape sequence
+    const char *start = buffer;
+    const char *end = first_invalid_escape;
+    int lineno = t->lineno;
+    int col_offset = t->col_offset;
+    while (start < end) {
+        if (*start == '\n') {
+            lineno++;
+            col_offset = 0;
+        }
+        else {
+            col_offset++;
+        }
+        start++;
+    }
+
+    // Count the number of quotes in the token
+    if (lineno == t->lineno) {
+        int quote_count = 0;
+        char* tok = PyBytes_AsString(t->bytes);
+        for (int i = 0; i < PyBytes_Size(t->bytes); i++) {
+            if (tok[i] == '\'' || tok[i] == '\"') {
+                quote_count++;
+            } else {
+                break;
+            }
+        }
+
+        col_offset += quote_count;
```

Comment:
Will it be correct for multiline output? I suggest to add a few more tests.

Reply:
There are tests already if I am not mistaken

Reply:
> There are tests already if I am not mistaken

Finally got some time to test this out, you was right, it performs great on multiline (offset is correct and isn't being affected by quotes in the start), but luckily I discovered some misbehaviour during testing.